test/network/web_socket_peer_test.dart

```diff
@@ -94,6 +102,50 @@ void main() {
       expect(tasks.length, equals(1));
       expect(tasks.first, task);
     });
+
+    test('should sync lists with connecting client', () async {
+      final list = TaskList(
+        title: 'Fun things to do',
+        id: '16ca13c-9021-4986-ab97-2d89cc0b3fce',
+      );
+      final message = <String, dynamic>{
+        '516ca13c-9021-4986-ab97-2d89cc0b3fce': {
+          'hlc':
+              '2021-06-04T07:37:08.946Z-0000-d5726a08-2107-49c0-8b06-167e57f96301',
+          'value': list.toJson(),
+        },
+      };
+
+      final peerLocation =
+          PeerLocation('ws://localhost:${await identityService.port}');
+      final client = WebSocketClient.connect(peerLocation.uri);
+      var completer = Completer();
+      client.dataStream.listen((data) {
+        completer.complete(data);
+      });
+
+      client.send(jsonEncode(Packet(
+        'TaskListsMessage',
+        object: TaskListsMessage(
+          jsonEncode(message),
+          requestReply: true,
+        ).toJson(),
+      )));
+      final serverData = await completer.future
+          .timeout(Duration(seconds: 5), onTimeout: () => null);
+      expect(
+        serverData,
+        isNot(equals(null)),
+        reason: 'server did not answer within 5s',
+      );
+
+      final lists = await taskListsService.lists;
+
+      expect(lists.length, equals(2));
+      expect(lists[0], list);
+      list.isShared = true;
+      expect(lists[1], list);
```

Comment:
Where does the second list come from and why are both lists equal to `list`?

Reply:
i am not quite sure if this is supposed to be like that but when `mergeCrdtJson` is called it will update the `isShared` to true calling `upsert` and thats why there are two entries.

in the second `equal` the list.isShared is true